src/check.py

```diff
@@ -89,14 +138,47 @@ def _update_action_problem(action: UpdateAction) -> Problem | None:
     return problem
 
 
-def conflicts(actions: Iterable[AnyAction]) -> Iterator[Problem]:
+def conflicts(
+    actions: Iterable[AnyAction], repository: Client, user_inputs: UserInputs
+) -> Iterator[Problem]:
     """Check whether actions have any content conflicts.
 
     Args:
         actions: The actions to check.
+        repository: Client for repository interactions.
+        user_inputs: Configuration from the user.
 
     Yields:
         A problem for each action with a conflict
     """
+    track_paths_with_diff = TrackPathsWithDiff()
+    commit_tagged = repository.is_same_commit(
+        tag=constants.DISCOURSE_AHEAD_TAG, commit=user_inputs.commit_sha
+    )
+
     update_actions = filter(_is_update_action, actions)
-    yield from filter(None, (_update_action_problem(action) for action in update_actions))
+    tracked_update_actions = side_effect(track_paths_with_diff.process, update_actions)
```

Comment:
very cool!

Reply:
I kinda disagree. :)

Why not using something like
```
tracked_update_actions = TrackPathsWithDiff.from(update_actions)
```
and avoid mutating object attributes? Also, if that's the use-case we need an extra dependency ('more_itertool`), I would really skip this logic

Reply:
As discussed above, this has been removed